pandas/tests/test_index.py

```diff
@@ -1548,54 +1604,7 @@ def test_groupby(self):
         exp = {1: [0, 1], 2: [2, 3, 4]}
         tm.assert_dict_equal(groups, exp)
 
-    def test_equals_op(self):
-        # GH9947
-        index_a = Index(['foo', 'bar', 'baz'])
-        index_b = Index(['foo', 'bar', 'baz', 'qux'])
-        index_c = Index(['foo', 'bar', 'qux'])
-        index_d = Index(['foo'])
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == index_b
-        assert_numpy_array_equivalent(index_a == index_a, np.array([True, True, True]))
-        assert_numpy_array_equivalent(index_a == index_c, np.array([True, True, False]))
-
-        # test comparisons with numpy arrays
-        array_a = np.array(['foo', 'bar', 'baz'])
-        array_b = np.array(['foo', 'bar', 'baz', 'qux'])
-        array_c = np.array(['foo', 'bar', 'qux'])
-        array_d = np.array(['foo'])
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == array_b
-        assert_numpy_array_equivalent(index_a == array_a, np.array([True, True, True]))
-        assert_numpy_array_equivalent(index_a == array_c, np.array([True, True, False]))
-
-        # test comparisons with Series
-        series_a = Series(['foo', 'bar', 'baz'])
-        series_b = Series(['foo', 'bar', 'baz', 'qux'])
-        series_c = Series(['foo', 'bar', 'qux'])
-        series_d = Series(['foo'])
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == series_b
-        assert_numpy_array_equivalent(index_a == series_a, np.array([True, True, True]))
-        assert_numpy_array_equivalent(index_a == series_c, np.array([True, True, False]))
-
-        # cases where length is 1 for one of them
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == index_d
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == series_d
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            index_a == array_d
-        with tm.assertRaisesRegexp(ValueError, "Series lengths must match"):
-            series_a == series_d
-        with tm.assertRaisesRegexp(ValueError, "Lengths must match"):
-            series_a == array_d
-
-        # comparing with scalar should broadcast
-        assert_numpy_array_equivalent(index_a == 'foo', np.array([True, False, False]))
-        assert_numpy_array_equivalent(series_a == 'foo', np.array([True, False, False]))
-        assert_numpy_array_equivalent(array_a == 'foo', np.array([True, False, False]))
-
+    def test_equals_op_multiindex(self):
```

Comment:
@jreback I'm able to include `MultiIndex` testing in the `test_equals_op` function above

everything works the same way except for one wrinkle, when testing the "broadcast when comparing with a scalar" rule:

```
In [1]: import pandas as pd

In [2]: mi = pd.MultiIndex.from_tuples([(1, 2), (4, 5), (8, 9)])

In [3]: mi
Out[3]:
MultiIndex(levels=[[1, 4, 8], [2, 5, 9]],
           labels=[[0, 1, 2], [0, 1, 2]])

In [4]: mi[0]
Out[4]: (1, 2)

In [5]: mi == mi[0]
Out[5]: False
```

Note that instead of broadcasting it thinks that `mi[0]` i.e. `(1, 2)` is an array of length 2, and the comparison in that case defaults to comparing the underlying numpy arrays, and therefore it returns `False`